replica/table.cc

```diff
@@ -856,7 +863,7 @@ compaction_group* table::single_compaction_group_if_available() const noexcept {
 }
 
 compaction_group* table::get_compaction_group(size_t id) const noexcept {
-    return storage_groups()[id]->main_compaction_group().get();
+    return &compaction_groups().front();
```

Comment:
why ignore the id arg?

Reply:
Shouldn't we still get storage_groups()[id]
and if it's not null return the address of its main compaction_group, else return nullptr?

@raphaelsc , btw, how does that work with splits?

Reply:
I will push a change with the assumption now, but please someone confirm/deny